src/metric/ackbag.rs

```diff
@@ -0,0 +1,104 @@
+use std::sync::{Arc, Mutex};
+use std::{thread, time};
+use util;
+use uuid::Uuid;
+
+#[derive(Clone)]
+pub struct SyncAckProps {
+    acks_received: usize
+}
+
+impl SyncAckProps {
+    pub fn new() -> Self {
```

Comment:
Recommend implementing Default for SyncAckProps versus having a blank new constructor.